plugin/rename.py

```diff
@@ -154,29 +153,35 @@ def run(
         edit: sublime.Edit,
         new_name: str = "",
         placeholder: str = "",
+        session_name: str | None = None,
         event: dict | None = None,
         point: int | None = None
     ) -> None:
         listener = self.get_listener()
         if listener:
             listener.purge_changes_async()
         location = get_position(self.view, event, point)
-        prepare_provider_session = self.best_session("renameProvider.prepareProvider")
+        prepare_provider_session = self._get_prepare_rename_session(point, session_name)
         if new_name or placeholder or not prepare_provider_session:
             if location is not None and new_name:
-                self._do_rename(location, new_name)
+                self._do_rename(location, new_name, prepare_provider_session)
                 return
             # Trigger InputHandler manually.
             raise TypeError("required positional argument")
         if location is None:
             return
         params = cast(PrepareRenameParams, text_document_position_params(self.view, location))
         request = Request.prepareRename(params, self.view, progress=True)
+        prepare_provider_session_name = prepare_provider_session.config.name
```

Comment:
Maybe a bit pedantic, but this variable could be just `session_name`. Otherwise it might imply that there are different session names for prepare provider and rename provider, but this is exactly what should be prevented with this.

Similarly we could/should just use `session` instead of `prepare_provider_session` above.